app.py

```diff
@@ -1 +1,521 @@
+            return coordinates
+
+        response = requests.get(url, params=params)
+
+        if response.status_code == 200:
+            response_json = response.json()
+
+            for feature in response_json['features']:
+                geometry = feature['geometry']
+                geometry_type = geometry['type']
+
+                coordinates = []
+                if geometry_type == 'Polygon':
+                    coordinates = _convert_polygon(geometry['coordinates'])
+                elif geometry_type == 'MultiPolygon':
+                    for polygon in geometry['coordinates']:
+                        coordinates.append(_convert_polygon(polygon))
+                else:
+                    logging.warning((
+                        f'Ignoring unknown geometry type: {geometry_type}. '
+                        f'(id: {feature["id"]})'
+                    ))
+
+                feature['geometry']['coordinates'] = coordinates
+
+        return response_json
+
+    def get_library_hours(self):
+        """
+        Get library open hours via library API
+        """
+        config = self.config['locations']['library']
+
+        headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/vnd.kiosks.v1'
+        }
+        body = {'dates': []}
+
+        for day in range(7):
+            week_day = self.today + timedelta(days=day)
+            body['dates'].append(utils.to_date(week_day))
+
+        response = requests.post(config['url'], headers=headers, json=body)
+
+        if response.status_code == 200:
+            open_hours = {}
+
+            for day in range(7):
+                week_day = self.today + timedelta(days=day)
+                open_hours[utils.to_date(week_day)] = []
+
+            for key, value in response.json().items():
+                datetime_format = '%Y-%m-%d %I:%M%p'
+                date = value['sortable_date']
+                begin = utils.format_library_hour(value['open'])
+                close = utils.format_library_hour(value['close'])
+
+                start = datetime.strptime(f'{date} {begin}', datetime_format)
+                end = datetime.strptime(f'{date} {close}', datetime_format)
+
+                open_hours[date] = {
+                    'summary': None,
+                    'uid': None,
+                    'start': utils.to_utc_string(start),
+                    'end': utils.to_utc_string(end),
+                    'sequence': None,
+                    'recurrenceId': None,
+                    'lastModified': None
+                }
+            return open_hours
+
+    def generate_json_resources(self):
+        """
+        Generate resources and write to JSON files
+        """
+        base_url = self.config['locations_api']['url']
+        facil_locations = self.get_facil_locations()
+        gender_inclusive_restrooms = self.get_gender_inclusive_restrooms()
+        arcGIS_geometries = self.get_arcGIS_geometries()
```

Comment:
Could facil locations, GI restrooms, and arcGIS geometries data be fetched asynchronously?

Reply:
We could, but it won't improve the performance significantly since these three data sources are pretty fast. Therefore, consider the ease of code readability, I guess I lean to keep the codes like this for now. What do you think?

Reply:
Ok, sounds good